### PR TITLE
Issue#36 Composite avro schemas implementation

### PR DESCRIPTION
**Pull request by satishd**

### CODE REVIEW

File: schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/SchemaResolver.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.schemaregistry;
+
+import com.hortonworks.registries.schemaregistry.errors.InvalidSchemaException;
+import com.hortonworks.registries.schemaregistry.errors.SchemaNotFoundException;
+
+/**
```
Comment (harshach) [inline schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/SchemaResolver.java]: can you add some doc explaining about the purpose of this interface

File: schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
```
Comment (harshach) [inline schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaResolver.java]: doc here as well.

File: schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/SchemaResolver.java (hunk 1 above)
Comment (satishd) [inline schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/SchemaResolver.java]: @harshach Added doc about the interface and implementation.

**Comment by satishd** [comment]
@harshach Addressed review comments.

**Pull request merged**